### PR TITLE
Added transactions to the block chain builder

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description
The simple block chain thats supposed to be used for testing only created new blocks with coinbase UTXO. It can now spend UTXO's with the each block.

## Motivation and Context
This allows for a much clearer blockchain to do testing on. 

## How Has This Been Tested?
Unit test looking at the chain that was constructed. The unit test is currently disabled because of #473 

## Types of changes
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by SWvheerden** [comment]
fixes #388

File: base_layer/core/src/block.rs [hunk 1]
```diff
@@ -67,10 +67,11 @@ impl Block {
 }
 
 // todo this probably need to move somewhere else
-/// This function will create the correct amount for the coinbase given the block height
+/// This function will create the correct amount for the coinbase given the block height, it will provide the answer in
```
Comment (CjS77) [inline base_layer/core/src/block.rs]: See `emission.rs` - let's incorporate that sooner rather than later

File: base_layer/core/src/block.rs [hunk 2]
```diff
@@ -67,10 +67,11 @@ impl Block {
...
+/// nanoTari
```
Comment (CjS77) [inline base_layer/core/src/block.rs]: Smallest denomination of Tari is µTari - I forget which RFC made that call

**Comment by SWvheerden** [comment]
Porting the block u64 to MicroTari is getting large, I am going to PR that separately.

File: base_layer/core/src/block.rs (hunk 2 above)
Comment (SWvheerden) [reply base_layer/core/src/block.rs]: Going to address this in a next pr refactoring u64 to MicroTari for amounts

File: base_layer/core/src/block.rs (hunk 1 above)
Comment (SWvheerden) [reply base_layer/core/src/block.rs]: Going to address this in a next pr refactoring u64 to MicroTari for amounts

**Pull request merged**